src/backend/distributed/executor/multi_server_executor.c

```diff
@@ -56,9 +56,11 @@ JobExecutorType(MultiPlan *multiPlan)
 	if (taskCount > 0)
 	{
 		Task *firstTask = (Task *) linitial(workerTaskList);
+		TaskType taskType = firstTask->taskType;
```

Comment:
The code block below `else if (executorType == MULTI_EXECUTOR_ROUTER)` should be safe  to remove.